PWGLF/Tasks/cascadecorrelations.cxx

```diff
@@ -0,0 +1,234 @@
+#include <cstdlib>
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoAHelpers.h"
+#include "ReconstructionDataFormats/Track.h"
+#include "Common/Core/RecoDecay.h"
+#include "Common/Core/trackUtilities.h"
+#include "PWGLF/DataModel/LFStrangenessTables.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/Centrality.h"
+#include "Common/DataModel/PIDResponse.h"
+#include <TFile.h>
+#include <TH2F.h>
+#include <TProfile.h>
+#include <TLorentzVector.h>
+#include <TPDGCode.h>
+#include <TDatabasePDG.h>
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using std::array;
+
+// use parameters + cov mat non-propagated, aux info + (extension propagated)
+using FullTracksExt = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksCov, aod::TracksDCA>;
+using FullTracksExtIU = soa::Join<aod::TracksIU, aod::TracksExtra, aod::TracksCovIU, aod::TracksDCA>;
+using FullTracksExtWithPID = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksCov, aod::TracksDCA, aod::pidTPCPi, aod::pidTPCKa, aod::pidTPCPr>;
+using FullTracksExtIUWithPID = soa::Join<aod::TracksIU, aod::TracksExtra, aod::TracksCovIU, aod::TracksDCA, aod::pidTPCPi, aod::pidTPCKa, aod::pidTPCPr>;
+
+// Add a column to the cascdataext table: IsSelected.
+// 0 = not selected, 1 = Xi, 2 = Omega
+namespace o2::aod
+{
+namespace cascadeflags
+{
+DECLARE_SOA_COLUMN(IsSelected, isSelected, int); //~!
+} // namespace cascadeflags
+DECLARE_SOA_TABLE(CascadeFlags, "AOD", "CASCADEFLAGS", //!
+                  cascadeflags::IsSelected);
+using CascDataExtSelected = soa::Join<CascDataExt, CascadeFlags>;
+} // namespace o2::aod
+
+struct cascadeSelector {
+  Produces<aod::CascadeFlags> cascflags;
+
+  // histo's
+  HistogramRegistry registry{
+    "registry",
+    {
+      {"hMassXiMinus", "hMassXiMinus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+    },
+  };
+
+  void process(soa::Join<aod::Collisions, aod::EvSels>::iterator const& collision, aod::CascDataExt const& Cascades, aod::V0sLinked const&, aod::V0Datas const&, FullTracksExtIUWithPID const&)
+  {
+    for (auto& casc : Cascades) {
+      auto v0 = casc.v0_as<o2::aod::V0sLinked>();
+      if (!(v0.has_v0Data())) {
+        cascflags(0);
+        continue; // reject if no v0data
+      }
+      auto v0data = v0.v0Data();
+
+      // Let's try to do some PID
+      // these are the tracks:
+      auto bachTrack = casc.bachelor_as<FullTracksExtIUWithPID>();
+      auto posTrack = v0data.posTrack_as<FullTracksExtIUWithPID>();
+      auto negTrack = v0data.negTrack_as<FullTracksExtIUWithPID>();
+
+      // PID
+      // Lambda check
+      if (casc.sign() < 0) {
+        // Proton check:
+        if (TMath::Abs(posTrack.tpcNSigmaPr()) > 3) {
+          cascflags(0);
```

Comment:
I suggest using a configurable for the cut of the nSigmaTPC (instead of 3 as hard-coded number). In this way you will be able to modify your selection without recompiling the code.

Reply:
The configurable could be different for different particles species to allow for more flexibility